torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -46,6 +47,55 @@ RRefForkData fromPyTuple(const py::tuple& pyTuple) {
 
   return RRefForkData(ownerId, rrefId, forkId, parent, typeStr);
 }
+
+TypePtr tryTypePyObjToTypePtr(const py::object& type_value) {
```

Comment:
I am thinking we should just merge `tryTypePyObjToTypePtr` with `decidePyObjJitType`, these two functions are quite confusing and take a while to know what they are really doing.

Maybe something like `tryToInferTypeFromPyObject(const py::object& value, const py::object& type_hint)`

Reply:
Merged.